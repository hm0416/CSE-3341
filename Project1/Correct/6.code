program
	add(ref a, b) begin
		a = a+b;
		ref n;
		b = ref n;
	endfunc
begin
	ref x;
	x = new;
	x = 1;
	ref y;
	y = new;
	y = 1+x;
	begin add(x, y);
	output x;
end