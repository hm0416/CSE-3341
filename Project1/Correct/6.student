gc:1
gc:2
3
gc:0
