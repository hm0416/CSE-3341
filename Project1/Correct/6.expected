PROGRAM
INT
ID[X]
COMMA
ID[Y]
COMMA
ID[Z]
SEMICOLON
INPUT
ID[X]
COMMA
ID[Y]
COMMA
ID[Z]
SEMICOLON
WHILE
NEGATION
LPAREN
ID[X]
ASSIGN
CONST[0]
RPAREN
BEGIN
OUTPUT
ID[Y]
SEMICOLON
WHILE
ID[Y]
LESS
CONST[10]
BEGIN
ID[Y]
ASSIGN
ID[Y]
ADD
CONST[1]
SEMICOLON
ENDWHILE
SEMICOLON
OUTPUT
ID[Y]
SEMICOLON
ID[Y]
ASSIGN
ID[Z]
ID[X]
ASSIGN
ID[X]
SUB
CONST[1]
SEMICOLON
ENDWHILE
SEMICOLON
END
